src/hooks/useSafeAreaInsets.ts

```diff
@@ -1,13 +1,19 @@
 import type {EdgeInsets} from 'react-native-safe-area-context';
 // eslint-disable-next-line no-restricted-imports
 import {useSafeAreaInsets as useSafeAreaInsetsInternal} from 'react-native-safe-area-context';
+import useKeyboardState from './useKeyboardState';
 import useStyleUtils from './useStyleUtils';
 
 function useSafeAreaInsets(): EdgeInsets {
     const StyleUtils = useStyleUtils();
     const insets = useSafeAreaInsetsInternal();
+    const {isKeyboardShown} = useKeyboardState();
```

Comment:
hmm, at the moment i don't experience any jumps/flickers. I don't know about the performance of this hook either.

Currently this change fixes too big paddings when opening the keyboard, because the safe area paddings are added, even though there is no safe area, when the keyboard is active.

In some cases, the safe area padding is applied multiple times, causing extra large paddings. This and other issues could be fixed in https://github.com/Expensify/App/pull/53229

Reply:
reverted this change @kirillzyusko